clippy_lints/src/lib.rs

```diff
@@ -654,7 +653,13 @@ pub fn register_lints(store: &mut rustc_lint::LintStore, conf: &'static Conf) {
                 .collect(),
         ))
     });
-    store.register_early_pass(|| Box::<utils::format_args_collector::FormatArgsCollector>::default());
+    let format_args_storage = FormatArgsStorage::default();
+    let format_args = format_args_storage.clone();
+    store.register_early_pass(move || {
+        Box::new(utils::format_args_collector::FormatArgsCollector::new(
+            format_args.clone(),
+        ))
+    });
```

Comment:
As this is a collection lint and the only one, it's a bit weird that it lives just between all the other registrations. Two things would make more sense: Run it before its first usage or register it before all other lints. I think the later would make the most sense. Can you move it before all lint registration and add a comment that no other regular lint pass should come before it?

Reply:
Moved it up next to the metadata collector, seems clear enough it's separate